golden_hammer/module.go

```diff
@@ -0,0 +1,275 @@
+func (h *GoldenHammer) registerServiceBaseURLs() error {
+	documents, err := h.listDocumentToFix()
+	if err != nil {
+		return err
+	}
+	for _, document := range documents {
+		var endpointToRegister url.URL
+		serviceEndpoint := getServiceEndpoint(document, transport.NutsCommServiceType)
+		if didservice.IsServiceReference(serviceEndpoint) {
+			// Care organization DID document, register service pointing to vendor DID.
+			parentDID, err := didservice.GetDIDFromURL(serviceEndpoint)
+			if err != nil {
+				// Invalid NutsComm reference, skip
+				log.Logger().WithError(err).
+					Debugf("Invalid NutsComm reference on DID %s: %s", document.ID, serviceEndpoint)
+				continue
+			}
+			// Only if the referenced document actually contains the service
+			if h.resolveContainsService(parentDID, types.BaseURLServiceType) {
+				log.Logger().Debugf("Could not resolve '%s' service in referenced (NutsComm) DID document (did=%s), skipping fix for DID: %s", types.BaseURLServiceType, parentDID.String(), document.ID)
+				continue
+			}
+			// All us OK
+			endpointToRegister = didservice.MakeServiceReference(parentDID, types.BaseURLServiceType).URL
+		} else {
+			// Vendor DID document, register resolved identifier
+			identifier, err := h.tryResolveURL(document.ID)
+			if identifier == nil || err != nil {
+				log.Logger().WithError(err).
+					Debugf("Could not resolve vendor's services base URL for DID %s, skipping fix", document.ID)
+				continue
+			}
+			endpointToRegister = *identifier
+		}
+		_, err := h.didmanAPI.AddEndpoint(h.ctx, document.ID, types.BaseURLServiceType, endpointToRegister)
+		if err != nil {
+			log.Logger().WithError(err).
+				Errorf("Unable to register DID services base URL (did=%s): %s", document.ID, endpointToRegister.String())
+		} else {
+			h.fixedDocumentDIDs[document.ID.String()] = true
+			log.Logger().Infof("Registered DIDs services base URL (did=%s): %s", document.ID, endpointToRegister.String())
+		}
+	}
+	return nil
+}
+
+func (h *GoldenHammer) listDocumentToFix() ([]did.Document, error) {
+	dids, err := h.documentOwner.ListOwned(h.ctx)
+	if err != nil {
+		return nil, err
+	}
+	var documents []did.Document
+	for _, id := range dids {
+		if h.fixedDocumentDIDs[id.String()] {
+			// Already fixed
+			continue
+		}
+		document, _, err := h.docResolver.Resolve(id, nil)
+		if err != nil {
+			if !didservice.IsFunctionalResolveError(err) {
+				log.Logger().WithError(err).Infof("Can't resolve DID document, skipping fix (did=%s)", id)
+			}
+			continue
+		}
+		if containsService(*document, types.BaseURLServiceType) {
+			h.fixedDocumentDIDs[id.String()] = true
+			continue
+		}
+		// This document needs fixing
+		documents = append(documents, *document)
+	}
+	// Sort: since care organization DIDs refer to vendor DIDs through NutsComm service,
+	// vendor DIDs should be fixed first. Meaning DIDs without NutsComm service should go first.
+	sort.Slice(documents, func(i, j int) bool {
```

Comment:
again, vendors have a NutsComm service. Only diff is that they have a concrete endpoint not a ref.

Reply:
fixed